### PR TITLE
expectNextPF with timeout #21314

### PR DESCRIPTION
**Pull request by manonthegithub**
In #21314 is described my problem with compilation, but that may be my local configuration problem so I created this PR, please advise me if I'm wrong.

### CODE REVIEW

**Comment by drewhk** [comment]
OK TO TEST

**Comment by manonthegithub** [comment]
As I see something was wrong with checking out from git, could you restart build?

**Comment by drewhk** [comment]
PLS BUILD

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by manonthegithub** [comment]
the problem i described in #21314  was not my local config problem... so what could you suggest guys? I just can't get why adding new overloaded function breaks type inference for previous version and why it compiled before?

File: akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala [hunk 1]
```diff
@@ -528,23 +528,46 @@ object TestSubscriber {
...
+     */
+    def expectNextPF[T](f: PartialFunction[Any, T]): T =
+      expectNextPF(Duration.Undefined, f)
+
+    /**
+     * Wait for the next element and test it with partial function.
+     *
+     * @param max wait no more than max time, otherwise throw AssertionError
+     */
+    def expectNextPF[T](max: Duration, f: PartialFunction[Any, T]): T =
```
Comment (manonthegithub) [inline akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala]: yeah, I tried, it isn't working.

**Comment by manonthegithub** [comment]
I made some research and community chatting and as I suspected it is bug in scala compiler. I have created [issue in scala project](https://issues.scala-lang.org/browse/SI-10074), and it seems not to be fixed in near future. 
I'm sorry for being inactive, just was very busy. 
I think the only way to solve it is to have functions with different names.

**Comment by patriknw** [comment]
ok, strange. Then we can perhaps change it to something similar as we have in actor Testkit: https://github.com/akka/akka/blob/master/akka-testkit/src/main/scala/akka/testkit/TestKit.scala#L402

That would break binary compatibility but testkits are excluded from that rule. 
However, it's would not be source compatible. Not so nice, but might be an alright change?
What do others think?

**Comment by patriknw** [comment]
Closing due to inactivity

**Comment by manonthegithub** [comment]
I can implement this, just thought that we are waiting for others to say if it's ok to break source compatibility?

**Comment by patriknw** [comment]
I don't think we would like to break source compatibility, but otherwise we are open for suggestions.

**Comment by manonthegithub** [comment]
then you can just merge this PR, as mentioned before it's not possible to override those functions with additional timeout param due to bug in scala compiler, so there are two options to add timeouts: merge this not beautiful PR or break source compat and reimplement in better codestyle as you suggested:

>Then we can perhaps change it to something similar as we have in actor Testkit: https://github.com/akka/akka/blob/master/akka-testkit/src/main/scala/akka/testkit/TestKit.scala#L402
That would break binary compatibility but testkits are excluded from that rule.
However, it's would not be source compatible.

**Comment by patriknw** [comment]
I see, sorry for misunderstanding. Let's merge this after validation.

**Comment by manonthegithub** [comment]
Isn't it stuck? Tests passed, what other checks to be completed?

**Comment by patriknw** [comment]
Thanks @manonthegithub

**Comment by manonthegithub** [comment]
Thank you too!) @patriknw

**Pull request merged**